### PR TITLE
Datadog forwarder - Draft

### PR DESCRIPTION
**Pull request by worldtiki**
### :pencil: Description

Draft for a new forwarder: input is Zipkin, output is Datadog.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Builder image
-FROM adoptopenjdk:15_36-jdk-hotspot as build
```
Comment (worldtiki) [inline Dockerfile:2]: using this pr to bump to java 16 as well

File: src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/datadog/DatadogDomainConverter.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    /**
+     * Accepts a span in {@code Zipkin V2} format and returns a span in {@code Datadog} format.
+     */
+    public static DatadogSpan fromZipkinV2(zipkin2.Span zipkin) {
+        Integer error = isError(zipkin);
+        BigInteger parentId = convertId(zipkin.parentId());
+        var spanId = convertId(zipkin.id());
+        var traceId = convertId(zipkin.traceId());
+
+        return new DatadogSpan(
```
Comment (worldtiki) [inline src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/datadog/DatadogDomainConverter.java:46]: some todos here. not sure if we need to solve all of these now though.

**Comment by worldtiki** [comment]
didn't update the docs because this is just experimental.

File: Dockerfile (hunk 1 above)
Comment (nikos912000) [reply Dockerfile:2]: Can we update the references to the Java version in the docs? E.g. readme, GitHub Pages.

File: src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 3]
```diff
@@ -150,6 +150,36 @@
...
+    },
+    {
+      "name": "pitchfork.forwarders.datadog.host",
+      "type": "java.lang.String",
+      "description": "The host for the datadog collector. Defaults to localhost."
+    },
+    {
+      "name": "pitchfork.forwarders.datadog.port",
+      "type": "java.lang.Integer",
+      "description": "The port for the datadog collector. Defaults 8216."
```
Comment (nikos912000) [inline src/main/resources/META-INF/additional-spring-configuration-metadata.json:166]:
```suggestion
      "description": "The port for the datadog collector. Defaults to 8216."
```

**Review by nikos912000 — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -4,4 +4,4 @@ test:
 	./mvnw clean verify
 
 build:
-	docker build -t expediagroup/pitchfork:test . -f Dockerfile
+	docker build -t worldtiki/pitchfork:test123 . -f Dockerfile
```
Comment (nikos912000) [inline Makefile:7]: Is this intentional?

**Review by nikos912000 — DISMISSED** [review]

**Review by nikos912000 — APPROVED** [review]

**Pull request merged**